src/main/java/javaserver/Routes.java

```diff
@@ -1,40 +1,38 @@
 package javaserver;
 
+import java.io.File;
 import java.util.HashMap;
 
-public class Routes {
-	public static final String[] FILES = {"/file1", "/text-file.txt"};
-
-	private static final String[] SUPPORTED_ROOT_REQUESTS = {"GET"};
-	private static final String[] SUPPORTED_FORM_REQUESTS = {"GET", "POST", "PUT"};
-	private static final String[] SUPPORTED_METHOD_OPTIONS = {"GET", "HEAD", "POST", "OPTIONS", "PUT"};
-	private static final String[] SUPPORTED_FILE_REQUESTS = {"GET"};
+import javaserver.ResponseBuilders.*;
 
-	private static final String[] FOUND_REDIRECT_REQUESTS = {"GET"};
-
-	public static HashMap<String, String[]> supportedRouteRequests;
-	public static HashMap<String, String[]> foundRouteRequests;
+public class Routes {
 
-	public static void configure() {
-		supportedRouteRequests = new HashMap<String, String[]>();
-		supportedRouteRequests.put("/", SUPPORTED_ROOT_REQUESTS);
-		supportedRouteRequests.put("/form", SUPPORTED_FORM_REQUESTS);
-		supportedRouteRequests.put("/method_options", SUPPORTED_METHOD_OPTIONS);
-		for (String fileRoute : FILES) {
-			supportedRouteRequests.put(fileRoute, SUPPORTED_FILE_REQUESTS);
+	public static HashMap<String, ResponseBuilder> routeResponders = new HashMap<String, ResponseBuilder>();
+	static {
+		routeResponders.put("/", new DirectoryResponseBuilder());
+		routeResponders.put("/parameters", new ParameterResponseBuilder());
+		routeResponders.put("/method_options", new OptionResponseBuilder());
+		routeResponders.put("/form", new FormResponseBuilder());
+		routeResponders.put("/redirect", new RedirectResponseBuilder());
+		for (String file : getDirectoryListing("public")) {
+			routeResponders.put("/" + file, new FileResponseBuilder());
 		}
-
-		foundRouteRequests = new HashMap<String, String[]>();
-		foundRouteRequests.put("/redirect", FOUND_REDIRECT_REQUESTS);
 	}
 
-	public static String[] getOptions(String route) {
-		if (supportedRouteRequests.get(route) != null) {
-			return supportedRouteRequests.get(route);
-		} else if (foundRouteRequests.get(route) != null) {
-			return foundRouteRequests.get(route);
-		} else {
-			return null;
+	public static HashMap<String, String[]> routeOptions = new HashMap<String, String[]>();
```

Comment:
It's a little strange that you're building two HashMaps here with the same keys. You could consider passing the supported HTTP methods to each ResponseBuilder as an argument, or you could store them inside each class as constants.